input/input.go

```diff
@@ -0,0 +1,252 @@
+			return ErrPollingIntervalFormat
+		}
+	} else {
+		return ErrPollingIntervalFormat
+	}
+	return nil
+}
+
+//isNumeric checks if string is numeric or not
+func isNumeric(s string) bool {
+	_, err := strconv.ParseFloat(s, 64)
+	return err == nil
+}
+
+// AuditJob represents the auditing job for which data will be fetched from remote git repository.
+type AuditJob struct {
+	// Name of the auditjob. <REQUIRED>
+	Name string `yaml:"name" json:"name"`
+	// Polling interval of the audit job. <REQUIRED> Format: 10s , 10m , 10h , 1d
+	PollingInterval string `yaml:"polling_interval" json:"polling_interval"`
+	// Metadata if any <OPTIONAL>
+	Metadata interface{} `yaml:"metadata" json:"metadata"`
+	// Tags if any <OPTIONAL>
+	Tags `yaml:"tags,omitempty" json:"tags"`
+	// Output targets. <REQUIRED>
+	Output `yaml:"output" json:"output"`
+	// Repository type , options include github , gitlabs etc. <REQUIRED>
+	RepositoryType string `yaml:"repo_type" json:"repo_type"`
+	// RepositoryName represents the name of the repository. <REQUIRED>
+	RepositoryName string `yaml:"repo_name" json:"repo_name"`
+	// RepositoryConfig defines reposiroty config. <REQUIRED>
+	RepositoryConfig `yaml:"repo_config" json:"repo_config"`
+}
+
+// RepositoryConfig represents repostory configurations.
+type RepositoryConfig struct {
+	// RepositoryURL is the url for the repository. <REQUIRED>
+	RepositoryURL string `yaml:"repo_url" json:"repo_url"`
+	// RepositoryCredentials represents repository credentials. <REQUIRED>
+	RepositoryCredentials `yaml:"credentials" json:"credentials"`
+	//Branches represents branches that needs to be monitored. <OPTIONAL>
+	Branches []string `yaml:"branches" json:"branches"`
+}
+
+// RepositoryCredentials consists of credential needed to authenticate with repository.
+type RepositoryCredentials struct {
+	//Either username or email is <REQUIRED>
+	//Username of the repository.
+	Username string `yaml:"username" json:"username"`
+	//Email of the repository.
+	Email string `yaml:"email" json:"email"`
+	//AccessToken for accessing the github's APIs. <REQUIRED>
+	AccessToken string `yaml:"access_token" json:"access_token"`
+}
+
+// Output represents the target where data will be sent.
+type Output struct {
+	//TargetName consists of the target names to which auditjob data needs to be sent. <REQUIRED>
+	TargetName []string `yaml:"target_name" json:"target_name"`
+}
+
+// Tags are additional information that can be added in tags fields of output documents.
+type Tags map[string]string
+
+// Target represents the target where data will be published.
+type Target struct {
+	//Name of the target. <REQUIRED>
+	Name string `yaml:"name" json:"name"`
+	//Type of the target , possible values (elasticsearch , kafka-rest etc). <REQUIRED>
+	Type string `yaml:"type" json:"type"`
+	//TargetConfig consist of the target configurations.
+	TargetConfig `yaml:"config" json:"config"`
+}
+
+//TargetConfig is key value target configurations.
+type TargetConfig map[string]string
```

Comment:
Can this be inside Target struct? TargetConfig map[string]string.

Reply:
changed